backend/src/Database.hs

```diff
@@ -1,157 +1,79 @@
 module Database where
```

Comment:
The simplicity of this file is crazy now! I think all the complexity was basically shifted off to types in `Types.Database`, which also is somehow beautifully simple looking because of Template Haskell? Huge refactor win

Reply:
Heck yeah! The downside of TH is that it's basically a bespoke DSL every time, so you always have to learn the custom syntax of the library author. The major upside is that sometimes that syntax is terse, beautiful, and way nicer than manually writing the equivalent "real" Haskell.

The `persistent` library we're using here has a really nice documentation book. It's perhaps useful to check out this one section, which shows the TH for defining a `Person` table (very similar to what we're doing here), and then shows the actual equivalent Haskell code that is generated by the TH: https://www.yesodweb.com/book/persistent#persistent_code_generation